cmd/iq.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/spf13/pflag"
+)
+
+var (
+	iqUsername  string
+	iqToken     string
+	iqHost      string
+	stage       string
+	application string
+	maxRetries  int
+	lifecycle   *iq.Server
+)
+
+var iqCmd = &cobra.Command{
+	Use:   "iq",
+	Short: "iq is used for auditing your projects with Nexus IQ Server",
+	RunE: func(cmd *cobra.Command, args []string) (err error) {
+		fflags := cmd.Flags()
+
+		err = checkRequiredFlags(fflags)
+		if err != nil {
+			return
+		}
+
+		printHeader()
+		logger = logrus.New()
+		switch verbose {
+		case 1:
+			logger.Level = logrus.InfoLevel
+		case 2:
+			logger.Level = logrus.DebugLevel
+		case 3:
+			logger.Level = logrus.TraceLevel
+		default:
+			logger.Level = logrus.ErrorLevel
+		}
+
+		lifecycle = iq.New(logger,
+			iq.Options{
+				User:          iqUsername,
+				Token:         iqToken,
+				Server:        iqHost,
+				Application:   application,
+				Stage:         stage,
+				Tool:          "ahab-client",
+				Version:       buildversion.BuildVersion,
+				OSSIndexToken: ossIndexToken,
+				OSSIndexUser:  ossIndexUser,
+				DBCacheName:   "ahab-cache",
+				MaxRetries:    maxRetries,
+			})
+
+		var list []string
+		scanner := bufio.NewScanner(os.Stdin)
+		for scanner.Scan() {
+			list = append(list, scanner.Text())
+		}
+		if err = scanner.Err(); err != nil {
+			if err != nil {
+				return
+			}
+		}
+
+		res, err := lifecycle.AuditPackages(tryParseStdInListIQ(list, &operating), application)
+		if err != nil {
+			return
+		}
+
+		fmt.Println()
+		if res.IsError {
+			logger.WithField("res", res).Error("An error occurred with the request to IQ Server")
+			return fmt.Errorf("Uh oh! There was an error with your request to Nexus IQ Server")
+		}
+
+		if res.PolicyAction != "Failure" {
+			logger.WithField("res", res).Debug("Successful in communicating with IQ Server")
+			fmt.Println("Wonderbar! No policy violations reported for this audit!")
+			fmt.Println("Report URL: ", res.ReportHTMLURL)
+			os.Exit(0)
+		} else {
+			logger.WithField("res", res).Debug("Successful in communicating with IQ Server")
+			fmt.Println("Hi, Nancy here, you have some policy violations to clean up!")
+			fmt.Println("Report URL: ", res.ReportHTMLURL)
+			os.Exit(1)
+		}
+		return
+	},
+}
+
+func tryParseStdInListIQ(list []string, operating *string) (purls []string) {
+	var thing string
+	thing = *operating
+	switch thing {
+	case "debian":
+		logger.Trace("Chasing Debian")
+		var aptResult packages.Apt
+		aptResult.ProjectList = parse.ParseDpkgList(list)
+		purls = aptResult.ExtractPurlsFromProjectList(*operating)
+	case "alpine":
+		logger.Trace("Chasing Alpine")
+		var apkResult packages.Apk
+		apkResult.ProjectList = parse.ParseApkShow(list)
+		purls = apkResult.ExtractPurlsFromProjectList(*operating)
+	default:
+		logger.Trace("Chasing Yum")
+		var yumResult packages.Yum
+		yumResult.ProjectList = parse.ParseYumListFromStdIn(list)
+		purls = yumResult.ExtractPurlsFromProjectList(*operating)
+	}
+	return
+}
```

Comment:
This code is currently duplicated between iq and oss we should probably pull it out and share it.

Reply:
It is SLIGHTLY shared, they call a different function, but I agree we can come up with a better thing :)

Reply:
Simplified everything down in: d15053a